### PR TITLE
Audio Update

### PR DESCRIPTION
**Pull request by Juniverse**
1. Fixed a bug for not able to speak in multi characters.
2. Used SNR to replace amplitude for better noise detecting.

Related Tasks:
INTG-1551 INTG-1482  INTG-1434  INTG-1583  INTG-1624  INTG-1622  INTG-1600  INTG-1629  INTG-1630  INTG-1631

### CODE REVIEW

File: Assets/Inworld/Inworld.Assets/Prefabs/3D/ChatScreen.prefab [hunk 1]
```diff
@@ -63,6 +63,7 @@ MonoBehaviour:
     relation: 1
     trigger: 0
     longBubbleMode: 0
+  m_EmotionMap: {fileID: 0}
```
Comment (mattkwilson) [inline Assets/Inworld/Inworld.Assets/Prefabs/3D/ChatScreen.prefab:66]: Missing reference?

File: Assets/Inworld/Inworld.Native/AEC/InworldAECAudioCapture.cs [hunk 2]
```diff
@@ -124,6 +126,8 @@ protected byte[] FilterAudio(short[] inputData, short[] outputData, IntPtr aecHa
             {
                 for (int i = 0; i < inputData.Length * m_Recording.channels; i += k_NumSamples)
                 {
+                    if (i + k_NumSamples > inputData.Length) 
+                        break;
```
Comment (mattkwilson) [inline Assets/Inworld/Inworld.Native/AEC/InworldAECAudioCapture.cs:130]: I think this be simplified to just:
```
for (int i = 0; i <= inputData.Length - k_NumSamples; i += k_NumSamples)
{
```
The added lines seems to defeat the purpose of multiplying by `m_Recording.channels` in the for loops condition.

File: Assets/Inworld/Inworld.Samples.RPM/Scripts/InworldFacialAnimationRPM.cs [hunk 3]
```diff
@@ -245,7 +250,7 @@ IEnumerator _MorphEmotion()
                     string currIterName = m_Skin.sharedMesh.GetBlendShapeName(i);
                     float fCurrShapeWeight = m_Skin.GetBlendShapeWeight(i);
                     MorphState lastState = m_LastFacial?.morphStates.FirstOrDefault(morph => morph.morphName == currIterName);
-                    MorphState currState = m_CurrentFacial?.morphStates.FirstOrDefault(morph => morph.morphName == currIterName);
+                    MorphState currState = m_CurrentFacial?.morphStates.FirstOrDefault(morph => morph.morphName == currIterName) ;
```
Comment (mattkwilson) [inline Assets/Inworld/Inworld.Samples.RPM/Scripts/InworldFacialAnimationRPM.cs:253]: Remove Space

File: Assets/Inworld/Inworld.Samples.RPM/Scripts/UI/EmotionCanvas.cs [hunk 4]
```diff
@@ -75,17 +77,14 @@ public void SetMainStatus(int mainStatus)
         /// <param name="nSpaffCode">the spaffcode of the emotion.</param>
         public void MockServerEmoEvents(int nSpaffCode)
         {
-            if (!m_Character || string.IsNullOrEmpty(m_Character.ID))
-            {
-                InworldAI.LogError("Please wait until character initialized!");
+            if (InworldController.Client.Status == InworldConnectionStatus.Connected)
```
Comment (mattkwilson) [inline Assets/Inworld/Inworld.Samples.RPM/Scripts/UI/EmotionCanvas.cs:80]: Are we only allowing users to mock Server Events when they are disconnected?

**Review by mattkwilson — CHANGES_REQUESTED** [review]

File: Assets/Inworld/Inworld.Samples.RPM/Scripts/UI/EmotionCanvas.cs (hunk 4 above)
Comment (Juniverse) [reply Assets/Inworld/Inworld.Samples.RPM/Scripts/UI/EmotionCanvas.cs:80]: Yes, applying mock events during runtime would occur some weird issues.

File: Assets/Inworld/Inworld.Assets/Scripts/Sample/AudioCaptureTest.cs [hunk 5]
```diff
@@ -77,13 +86,18 @@ void _InitUI()
             }
     #endif
         }
-        
 
+        public override void Calibrate()
+        {
+            Debug.Log("HIT");
```
Comment (mattkwilson) [inline Assets/Inworld/Inworld.Assets/Scripts/Sample/AudioCaptureTest.cs]: Should this be here?

**Review by mattkwilson — APPROVED** [review]

**Pull request merged**